### PR TITLE
Search bug fixes and improvements

### PR DESCRIPTION
**Pull request by nagem**
- Fixed include all data checkbox
- Fixed Analysis project/session info
- Added `limit` query param to limit the results returned
- Added `api/search/field` endpoint for getting aggregations about values of a field.

### `api/search/field` Examples:
_________
Note: the result counts are currently for all data, not just the user's data. They are also approximate due to Elastic sharding logic.

#### Request:
```
POST /api/search/field HTTP/1.1
Host: docker.local.flywheel.io:8443
Content-Type: application/json
{
    "doc_type": "files",
    "field": "type"
}
```
#### Response:
```
[
  {
    "key": "nifti",
    "doc_count": 38
  },
  {
    "key": "dicom",
    "doc_count": 37
  },
  {
    "key": "qa",
    "doc_count": 6
  },
  {
    "key": "bval",
    "doc_count": 3
  },
  {
    "key": "bvec",
    "doc_count": 3
  },
  {
    "key": "image",
    "doc_count": 2
  },
  {
    "key": "source code",
    "doc_count": 1
  },
  {
    "key": "tabular data",
    "doc_count": 1
  }
]
```

#### Request:
```
POST /api/search/field HTTP/1.1
Host: docker.local.flywheel.io:8443
Content-Type: application/json

{
    "doc_type": "subject",
    "field": "code"
}
```
#### Response:
```
[
  {
    "key": "ex1",
    "doc_count": 1
  },
  {
    "key": "ex149",
    "doc_count": 1
  },
  {
    "key": "ex2878",
    "doc_count": 1
  },
  {
    "key": "ex2951",
    "doc_count": 1
  },
  {
    "key": "ex3967",
    "doc_count": 1
  },
  {
    "key": "ex4374",
    "doc_count": 1
  },
  {
    "key": "ex468",
    "doc_count": 1
  },
  {
    "key": "ex4784",
    "doc_count": 1
  },
  {
    "key": "ex6879",
    "doc_count": 1
  },
  {
    "key": "ex7236",
    "doc_count": 1
  }
]
```

#### Request:
```
POST /api/search/field HTTP/1.1
Host: docker.local.flywheel.io:8443
Content-Type: application/json
{
    "doc_type": "sessions",
    "field": "tags"
}
```
#### Response:
```
[
  {
    "key": "blue",
    "doc_count": 7
  },
  {
    "key": "green",
    "doc_count": 6
  },
  {
    "key": "black",
    "doc_count": 5
  },
  {
    "key": "red",
    "doc_count": 5
  },
  {
    "key": "yellow",
    "doc_count": 5
  },
  {
    "key": "white",
    "doc_count": 4
  },
  {
    "key": "grey",
    "doc_count": 3
  },
  {
    "key": "orange",
    "doc_count": 3
  },
  {
    "key": "pink",
    "doc_count": 3
  },
  {
    "key": "purple",
    "doc_count": 3
  }
]
```

### CODE REVIEW

**Comment by nagem** [comment]
@rentzso the `limit` query param is pretty naive, it just trims the results before sending them to the client. Let me know if you have any ideas for using Elastic's limit without affecting the search path results. (For example, looking for very specific files but the search for sessions only returns the top 200 sessions so the other N sessions aren't part of the next part of the search tree.)

**Comment by rentzso** [comment]
Yeah, the "limit" behavior is something we always punted on. We may use elasticsearch pagination support, but check the warning on the elasticsearch: https://www.elastic.co/guide/en/elasticsearch/guide/1.x/pagination.html

I may have an idea that can work (but that will require a relevant refactoring), but I would like to hear an elasticsearch consultant about our search system before starting any work on it (we were planning that, weren't we? CC @gsfr ).

**Comment by kofalt** [comment]
Chiming in here to say this looks neat :camera: :+1:

File: api/handlers/searchhandler.py [hunk 1]
```diff
@@ -168,3 +186,33 @@ def get_datatree(self):
...
+        if field_name.startswith('info'):
+            self.abort(400, 'Terms for info fields are not supported.')
+        if doc_type == 'subject' and field_name not in allowable_subject_agg_fields:
+            self.abort(400, 'Allowable fields for subject doc_type: {}.'.format(allowable_subject_agg_fields))
+
+        query = es_aggs(doc_type, field_name)
+        try:
+            # pylint disable can be removed after PyCQA/pylint#258 is fixed
+            es_results = config.es.search(index='scitran', body=query) # pylint: disable=unexpected-keyword-arg
+            ## elastic search results are wrapped in subkey ['hits']['hits']
```
Comment (rentzso) [inline api/handlers/searchhandler.py]: is this comment correct?

File: api/search/__init__.py [hunk 2]
```diff
@@ -35,6 +35,20 @@ def es_query(input_query, doc_type, min_score=0.5, additional_filter=None, sourc
...
 
+def es_aggs(doc_type, field_name):
+    query = {
+      "size": 0,
+      "aggs": {}
+    }
+
+    query['aggs'][field_name] = {
+        "terms": {
+            "field": '.'.join((doc_type, field_name, 'exact_'+field_name))
```
Comment (rentzso) [inline api/search/__init__.py]: I assume this is the right way of using `field_name` for aggregation. 
It is like the `exact_` field is a subfield, right?

**Comment by gsfr** [comment]
@rentzso Yes, I am in full support of bringing in a consultant to get us on the right track with ES. Let's try to limp along until @nagem returns from ElastiCon.

@nagem Please try to connect with potential consultants at the conference.

**Pull request merged**